lib/src/values/config_number.cc

```diff
@@ -31,4 +37,32 @@ namespace hocon {
     bool config_number::operator!=(const config_number &other) const {
         return !(*this == other);
     }
+
+    int config_number::int_value_range_checked(std::string path) {
```

Comment:
This is used for something way down the line. As I pointed out in my comments on why I originally removed config_int, config_int only returns its value as a long (to help make them more interchangeable, I guess). There are a couple of places where the Java decides to make a distinction between longs and ints, and this method is used to get the int value if possible. It's kinda weird, I know. As far as I can tell, the path has something to do with where in the document structure the number sits, but I honestly don't know how it works yet.